File: 12/solution.py
def getFactors(target):
	factors = []
	for i in xrange(1, int(target ** 0.5)):
		if target % i == 0:
			factors.append(i)
			factors.append(target / i)
	return factors

done = False
term = 0
triangle_number = 0
while not done:
	term += 1
	triangle_number += term
 	if len(getFactors(triangle_number)) > 500:
 		done = True

print triangle_number

File: 12/solution.js
function getFactors(target) {
	var factors = [];
	for (var i = 0; i < Math.sqrt(target); i += 1) {
		if (target % i === 0) {
			factors.push(i);
			factors.push(target / i);
		}
	}
	return factors;
}

var done = false;
var term = 0;
var triangle_number = 0;
while (!done) {
	term += 1;
	triangle_number += term;
	if (getFactors(triangle_number).length > 500) {
		done = true;
	}
}

console.log(triangle_number);